### ISSUE TITLE
Make VirtualBox port forwarding configurable

### ISSUE THREAD

**Issue by joshjdevl** [docker/machine#41]
Hi,

It would be helpful if we can configure the port forwarding when installing a new box.

Thanks

**Comment by ianneub** [comment]
:+1: I'd love to see this support. It would make it easier for my dev's to setup an environment and be able to copy/paste the URLs of their local machines around so that other's can see their work.

**Comment by ianneub** [comment]
Bonus points for being able to give it a range of ports to forward. Use cases: Our devs are working on multiple website projects at once and we want to pass the URLs of the local machine around to team leaders or what have you to inspect/critique the work.

**Comment by divyakumarjain** [comment]
:+1:

**Comment by MiLk** [comment]
:+1:

**Comment by charlesg3** [comment]
:+1:

**Comment by pnovotnak** [comment]
This would be badass. Otherwise, I don't know how `docker-machine` is calling `ssh`, but port forwarding  that way would also be fine by me.

**Comment by thelazyenginerd** [comment]
:+1:

**Comment by cdierkens** [comment]
Using a machine as a local server is a use case. I want to provide my developers with the ability to create and start a machine that can serve a web applications that can be hit from other machines on the network. What I DON'T want is have "Open virtual box GUI, do these steps, click this box, and now you have port forwarding. What I DO want is to provide a command "docker-machine create -d virtualbox -p 8080:8080 default"

**Comment by howdoicomputer** [comment]
Yeah, so I wrote this Fish function that I tied into some other functions to prepare the vm for my project. Maybe someone will find some use for it. 

```
function vbox_port -d 'Forwards VirtualBox port'
  set name $argv[1]
  set port $argv[2]

  VBoxManage modifyvm drought --natpf1 "$name,tcp,127.0.0.1,$port,,$port"
end
```

One problem I have is that I create my VM but when I put my laptop through a suspend cycle port forwarding breaks and I'm not sure why. The only fix I've found is removing the vm and then recreating it. The downside is that I have to pull down my containers again and that's kinda' puts a halt towards development on the go.

**Comment by bdesilva-appneta** [comment]
:+1:

**Comment by farums** [comment]
```sh
 BLUE='\033[1;34m'
 GREEN='\033[0;32m'
 RED='\033[0;32m'
 NC='\033[0m'
 VM=app-docker   
 
 if [ ! -z "$VBOX_MSI_INSTALL_PATH" ]; then
   VBOXMANAGE="${VBOX_MSI_INSTALL_PATH}VBoxManage.exe"
 else
   VBOXMANAGE="${VBOX_INSTALL_PATH}VBoxManage.exe"
 fi
echo -e "${BLUE}VBoxManage -v ${NC} = $("${VBOXMANAGE}" -v)"
VBoxManage modifyvm "${VM}" --natpf1 "my-web,tcp,127.0.0.1,8080,,80"
VBoxManage modifyvm "${VM}" --natpf1 "my-ssh,tcp,127.0.0.1,2200,,220"
```

**Comment by selvaraj-kuppusamy** [comment]
Make it usefull:
https://github.com/selvaraj-kuppusamy/port_forwading-virtualbox
